x-pack/plugins/apm/public/components/shared/ApmHeader/service_name_header/icon_popover.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { EuiIcon } from '@elastic/eui';
+import { EuiButtonEmpty, EuiPopover, EuiPopoverTitle } from '@elastic/eui';
+import React, { useState } from 'react';
+
+interface IconPopoverProps {
+  icon: string;
+  title: string;
+  children: React.ReactChild;
+}
+export function IconPopover({ icon, title, children }: IconPopoverProps) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const tooglePopover = () => {
+    setIsOpen((prevState) => !prevState);
+  };
+
+  return (
+    <EuiPopover
+      button={
+        <EuiButtonEmpty onClick={tooglePopover}>
+          <EuiIcon type={icon} size="l" color="black" />
+        </EuiButtonEmpty>
+      }
+      isOpen={isOpen}
+      closePopover={tooglePopover}
+    >
+      <EuiPopoverTitle>{title}</EuiPopoverTitle>
+      {children}
+    </EuiPopover>
```

Comment:
Would it be possible to set a min-width of 300px, both so there's enough room but also make the popover look and feel the same for either option the user picks?